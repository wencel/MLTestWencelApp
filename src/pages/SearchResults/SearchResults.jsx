import PropTypes from 'prop-types';
import SearchNavbar from 'components/SearchNavbar';
import { useEffect } from 'react';
import ItemsList from 'components/ItemsList';
import BreadCrums from 'components/BreadCrums';
import ContentContainer from 'components/ContentContainer';

const SearchResults = ({
  redirectSearchItems,
  requestSearchItems,
  items,
  search,
}) => {
  useEffect(() => {
    if (search) requestSearchItems(search);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [search]);
  return (
    <>
      <SearchNavbar
        onSearch={redirectSearchItems}
        initialSearchTerm={search ? search : ''}
      />
      <BreadCrums categories={items.data.categories} />
      <ContentContainer>
        <ItemsList
          items={items.data.items}
          loading={items.loading}
          error={items.error}
        />
      </ContentContainer>
    </>
  );
};

SearchResults.propTypes = {
  items: PropTypes.shape({
    data: PropTypes.shape({
      categories: PropTypes.array,
      items: PropTypes.array,
    }),
    error: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
    loading: PropTypes.bool,
  }),
  redirectSearchItems: PropTypes.func,
  requestSearchItems: PropTypes.func,
  search: PropTypes.string,
};

SearchResults.defaultProps = {
  redirectSearchItems: () => {},
  requestSearchItems: () => {},
};

export default SearchResults;
