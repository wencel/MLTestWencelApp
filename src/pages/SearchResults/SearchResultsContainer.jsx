import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';
import { useLocation } from 'react-router-dom';
import {
  redirectSearchItems,
  requestSearchItems,
} from 'reducers/item/itemActions';
import { itemsSelector } from 'reducers/item/itemSelectors';
import { getQueryTextParams } from 'utils/utils';
import SearchResults from './SearchResults';

const SearchResultsContainer = ({
  redirectSearchItems,
  requestSearchItems,
  items,
}) => {
  const search = getQueryTextParams(useLocation().search).search || '';
  return (
    <SearchResults
      redirectSearchItems={redirectSearchItems}
      requestSearchItems={requestSearchItems}
      items={items}
      search={search}
    />
  );
};

SearchResultsContainer.propTypes = {
  items: PropTypes.shape({
    data: PropTypes.shape({
      categories: PropTypes.array,
      items: PropTypes.array,
    }),
    error: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
    loading: PropTypes.bool,
  }),
  redirectSearchItems: PropTypes.func,
  requestSearchItems: PropTypes.func,
};

const mapStateToProps = state => {
  return {
    items: itemsSelector(state),
  };
};

const mapDispatchToProps = dispatch => ({
  redirectSearchItems: data => {
    dispatch(redirectSearchItems(data));
  },
  requestSearchItems: data => {
    dispatch(requestSearchItems(data));
  },
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SearchResultsContainer);
