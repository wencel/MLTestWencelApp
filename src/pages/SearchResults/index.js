import SearchResults from './SearchResultsContainer';
export default SearchResults;
