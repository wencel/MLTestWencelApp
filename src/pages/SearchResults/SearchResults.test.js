import React from 'react';
import { shallow } from 'enzyme';
import SearchResults from './SearchResults';

describe('SearchResults', () => {
  const redirectSearchItems = jest.fn();
  const requestSearchItems = jest.fn();
  const items = { data: {}, loading: false, error: '' };
  const search = 'test';
  const wrapper = shallow(
    <SearchResults
      redirectSearchItems={redirectSearchItems}
      requestSearchItems={requestSearchItems}
      items={items}
      search={search}
    />
  );

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('calls the requestSearchItems function when on mount', () => {
    // Testing the useEffect requires a timeout for it to work
    setTimeout(() => {
      expect(requestSearchItems).toHaveBeenCalledTimes(1);
    }, 0);
  });
});
