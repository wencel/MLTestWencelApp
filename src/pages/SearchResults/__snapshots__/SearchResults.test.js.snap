// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When it mounts the component renders correctly 1`] = `ShallowWrapper {}`;
